internal/list.go

```diff
@@ -11,3 +34,320 @@ func newListProxy(client *HazelcastClient, serviceName *string, name *string) (*
+func (list *ListProxy) AddItemListener(listener interface{}, includeValue bool) (registrationID *string, err error) {
+	request := ListAddListenerEncodeRequest(list.name, includeValue, false)
+	onItemEvent := func(itemData *Data, uuid *string, eventType int32) {
+		var item interface{}
+		if includeValue {
+			item, _ = list.ToObject(itemData)
+		}
+		member := list.client.ClusterService.GetMemberByUuid(*uuid)
+		itemEvent := NewItemEvent(list.name, item, eventType, member.(*Member))
+		if eventType == ITEM_ADDED {
+			if _, ok := listener.(core.ItemAddedListener); ok {
+				listener.(core.ItemAddedListener).ItemAdded(itemEvent)
+			}
+		} else if eventType == ITEM_REMOVED {
+			if _, ok := listener.(core.ItemRemovedListener); ok {
+				listener.(core.ItemRemovedListener).ItemRemoved(itemEvent)
+			}
+		}
+
+	}
+	eventHandler := func(clientMessage *ClientMessage) {
+		ListAddListenerHandle(clientMessage, func(itemData *Data, uuid *string, eventType int32) {
+			onItemEvent(itemData, uuid, eventType)
+		})
+	}
+	return list.client.ListenerService.registerListener(request, eventHandler,
+		func(registrationId *string) *ClientMessage {
+			return ListRemoveListenerEncodeRequest(list.name, registrationId)
+		}, func(clientMessage *ClientMessage) *string {
+			return ListAddListenerDecodeResponse(clientMessage).Response
+		})
+}
+
+func (list *ListProxy) Clear() (err error) {
+	request := ListClearEncodeRequest(list.name)
+	_, err = list.Invoke(request)
+	return err
+}
+
+func (list *ListProxy) Contains(element interface{}) (found bool, err error) {
+	if element == nil {
+		return false, errors.New(NIL_VALUE_IS_NOT_ALLOWED)
+	}
+	valueData, err := list.ToData(element)
+	if err != nil {
+		return false, err
+	}
+	request := ListContainsEncodeRequest(list.name, valueData)
+	responseMessage, err := list.Invoke(request)
+	if err != nil {
+		return false, err
+	}
+	found = ListContainsDecodeResponse(responseMessage).Response
+	return found, nil
+}
+
+func (list *ListProxy) ContainsAll(elements []interface{}) (foundAll bool, err error) {
+	if len(elements) == 0 {
+		return false, errors.New(NIL_VALUE_IS_NOT_ALLOWED)
+	}
+	elementsData, err := collection.ObjectToDataCollection(elements, list.client.SerializationService)
+	if err != nil {
+		return false, err
+	}
+	request := ListContainsAllEncodeRequest(list.name, elementsData)
+	responseMessage, err := list.Invoke(request)
+	if err != nil {
+		return false, nil
+	}
+	foundAll = ListContainsAllDecodeResponse(responseMessage).Response
+	return foundAll, nil
+}
+
+func (list *ListProxy) Get(index int32) (element interface{}, err error) {
+	request := ListGetEncodeRequest(list.name, index)
+	responseMessage, err := list.Invoke(request)
+	if err != nil {
+		return nil, err
+	}
+	element = ListGetDecodeResponse(responseMessage).Response
+	return element, nil
+}
+
+func (list *ListProxy) IndexOf(element interface{}) (index int32, err error) {
+	if element == nil {
+		return 0, errors.New(NIL_VALUE_IS_NOT_ALLOWED)
```

Comment:
It should return `HazelcastIllegalArgumentError`. 
In Java, it is like this:
`throw new IllegalArgumentException(format("argument '%s' can't be null", argName));`

Reply:
I think it is NullPointerException on java. 
did we introduce an equivalent of `NullPointerException` in go client?

Reply:
@sancar  There is `HazelcastIllegalArgumentError` in go client.